# Data Analysis on Heart Disease

## Introduction
One of the leading causes of death in the world and the leading cause in the United States, is heart disease and has been for many years. With the rate of them increasing worldwide in recent years, it is important to look at the leading causes of this disease and establish a relationship between them and its global increase. As a base for our analysis we are going to use the heart disease dataset available at Kaggle.

The dataset originates from 1988 and evaluates data from four different databases: Cleveland, Hungary, Switzerland and Long Beach V. While the dataset does have 76 distinct attributes, it only utilizes 14 of them. 

**Predictive question:**
How does the amount of cholesterol and the type of heart defect help us predict the diagnosis of heart disease?


In [9]:
### Run this cell before continuing.

library(repr)
library(tidyverse)
library(tidymodels)
options(repr.matrix.max.rows = 10)

**Reading the Data**

First read the raw data from GitHub that we uploaded.

In [10]:
heart_disease_data <- read_csv("https://raw.githubusercontent.com/yma24ma/dsci_009_43_gp/main/heart.csv")
heart_disease_data

Rows: 1025 Columns: 14
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
dbl (14): age, sex, cp, trestbps, chol, fbs, restecg, thalach, exang, oldpea...

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.


age sex cp trestbps chol fbs restecg thalach exang oldpeak slope ca thal
1    52  1   0  125      212  0   1       168     0     1.0     2     2  3   
2    53  1   0  140      203  1   0       155     1     3.1     0     0  3   
3    70  1   0  145      174  0   1       125     1     2.6     0     0  3   
4    61  1   0  148      203  0   1       161     0     0.0     2     1  3   
5    62  0   0  138      294  1   1       106     0     1.9     1     3  2   
⋮    ⋮   ⋮   ⋮  ⋮        ⋮    ⋮   ⋮       ⋮       ⋮     ⋮       ⋮     ⋮  ⋮   
1021 59  1   1  140      221  0   1       164     1     0.0     2     0  2   
1022 60  1   0  125      258  0   0       141     1     2.8     1     1  3   
1023 47  1   0  110      275  0   0       118     1     1.0     1     1  2   
1024 50  0   0  110      254  0   0       159     0     0.0     2     0  2   
1025 54  1   0  120      188  0   1       113     0     1.4     1     1  3   
     target
1    0     
2    0     
3    0     
4    0     
5    0     
⋮    ⋮     
1021 1     
1022 0     
1023 0     
1024 1     
1025 0

**Variables**

age: Age

sex: Sex

cp: Chest pain type (4 values)

trestbps: resting blood pressure

chol: serum cholestoral in mg/dl

fbs: fasting blood sugar > 120 mg/dl

restecg: resting electrocardiographic results (values 0,1,2)

thalacg: maximum heart rate achieved

exang: exercise induced angina

oldpeak: ST depression induced by exercise relative to rest

slope: the slope of the peak exercise ST segment

ca: number of major vessels (0-3) colored by flourosopy

thal: 0 = normal; 1 = fixed defect; 2 = reversable defect

target: diagnosis of heart disease

**Select Data**

We will now be using the `select()` function to select the row that we will be using in this analysis and make it into one table.

In [11]:
heart_disease_selected <- select(heart_disease_data, chol, thal, target)
heart_disease_selected

chol thal target
1    212  3    0     
2    203  3    0     
3    174  3    0     
4    203  3    0     
5    294  2    0     
⋮    ⋮    ⋮    ⋮     
1021 221  2    1     
1022 258  3    0     
1023 275  2    0     
1024 254  2    1     
1025 188  3    0

**Average of every column**

In [17]:
hd_average <- summarize(heart_disease_selected,
                            average_chol = mean(chol, na.rm = TRUE),
                            average_thal = mean(thal, na.rm = TRUE),
                            averag_target = mean(target, na.rm = TRUE))
hd_average

average_chol average_thal averag_target
1 246          2.323902     0.5131707